src/main/java/array/MainObject_Advance.java

```diff
@@ -173,35 +173,26 @@ public static void main(String[] args) {
         }
     }
 
-
+    //<editor-fold desc="Private functions">
     private static void showInfoBasic(Account account) {
         printSeparatedLine(50, "=");
-        System.out.println("Ho ten: " + account.getFirstName() + " " + account.getLastName());
-        System.out.println("Ngay tao: " + formatDate(account.getCreatedDate()));
+        System.out.println("Họ tên: " + account.getFirstName() + " " + account.getLastName());
+        System.out.println("Ngày tạo: " + formatDate(account.getCreatedDate()));
         System.out.println("Số tiền: " + formatCurrency(account.getCurrentBalance()));
         System.out.println("Loại kỳ hạn: " + account.getPeriod());
     }
 
-    private static void showInfo(Account account){
-     //   printSeparatedLine(50, "=");
-        System.out.println("Ho ten: " + account.getFirstName() + " " + account.getLastName());
-        System.out.println("Ngay tao: " + formatDate(account.getCreatedDate()));
-        System.out.println("Số tiền: " + formatCurrency(account.getCurrentBalance()));
-        System.out.println("Loại kỳ hạn: " + account.getPeriod());
-        System.out.println("Lịch sử giao dịch: ");
-        if (account.getHistories().size() == 0) {
-            System.out.println("Tài khoản chưa thực hiện bất kì giao dịch nào");
+    private static void showInfo(Account account, AccountHistory.Type type) {
+        List<AccountHistory> listAccountHistory = account.getHistories().stream().filter(transferIn ->
+                transferIn.getType() == type).collect(Collectors.toList());
+        showInfo(account, listAccountHistory);
+    }
 
-        }
-        for (AccountHistory history : account.getHistories()) {
-            dash();
-            System.out.println("Ngày tạo: " + formatDate(history.getCreatedDate(), "dd-MM-yyyy hh:mm:ss"));
-            System.out.println("Số tiền: " + formatCurrency(history.getBalance()));
-            System.out.println("Loại giao dịch: " + history.translatedType());
-        }
-        printSeparatedLine(50, "=");
+    private static void showInfo(Account account){
+        showInfo(account, account.getHistories());
     }
-    private static void showInfo(Account account, AccountHistory.Type type){
+
+    private static void showInfo(Account account, List<AccountHistory> listAccountHistory){
```

Comment:
Viết gọn lại để tránh duplicate code, hàm này chỉ nhận nhiệm vụ in ra thông tin tài khoản và lịch sử giao dịch đã đc chỉ định cần in ra (có thể chọn in hết hoặc in những lịch sử giao dịch đc chỉ định) 
Hàm này chỉ nhận vào cái cần in và in ra thôi, việc quyết định thông tin in ra hàm mà gọi hàm này sẽ quyết định.
-> cách này làm cho code tường minh hơn, mỗi hàm chỉ nên có 1 nhiệm vụ duy nhất thôi.